config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java

```diff
@@ -155,39 +154,23 @@ public void deleteConfigs(
                 .asException());
         return;
       }
-      Map<ConfigResourceContext, Value> valuesByContext =
-          request.getConfigsList().stream()
-              .map(
-                  requestedDelete ->
-                      Map.entry(this.getConfigResourceContext(requestedDelete), emptyValue()))
-              .collect(ImmutableMap.toImmutableMap(Entry::getKey, Entry::getValue));
-
-      List<UpsertedConfig> deletedConfigs =
-          configStore.writeAllConfigs(valuesByContext, getUserId());
 
+      Set<ConfigResourceContext> configResourceContexts =
+          request.getConfigsList().stream()
+              .map(this::getConfigResourceContext)
+              .collect(Collectors.toUnmodifiableSet());
+      List<ContextSpecificConfig> configs = configStore.getConfigs(configResourceContexts);
+      // delete the configs for the specified config resources.
+      configStore.deleteConfigs(configResourceContexts);
       responseObserver.onNext(
-          DeleteConfigsResponse.newBuilder()
-              .addAllDeletedConfigs(
-                  deletedConfigs.stream()
-                      .map(this::buildDeletedContextSpecificConfig)
-                      .collect(Collectors.toUnmodifiableList()))
-              .build());
+          DeleteConfigsResponse.newBuilder().addAllDeletedConfigs(configs).build());
```

Comment:
Do we need to return the entire set of deleted configs? Or, just the config ids?

Reply:
contract says the complete config. thats why made the change to not break the contract